### PR TITLE
return detail message from reponse body if authentication failed.

### PR DESCRIPTION
**Pull request by hopkings2008**
Dear maintainers,
This PR is to return the detailed message from response body when authentication failed.
Currently, if authentication failed, there is no detail message of why it failed, and user cannot know what to do next.
Add this mechanism, so that we can return the detail message and help user to make authentication pass and easy to use.

### CODE REVIEW

File: registry/client/errors.go [hunk 1]
```diff
@@ -47,7 +47,7 @@ func parseHTTPErrorResponse(r io.Reader) error {
 	return errors
 }
 
-func handleErrorResponse(resp *http.Response) error {
+func HandleErrorResponse(resp *http.Response) error {
```
Comment (stevvooe) [inline registry/client/errors.go]: Please document exported functions.

@dmcgowan Is this okay to export? Do we see any issues with that?

**Comment by hopkings2008** [comment]
@stevvooe , thanks for your checking and i have added the comments in that new exported function. That function is used by 'client' which is used by docker daemon. If we have that function, we can display the response body to end user. :)

**Comment by RichardScothern** [comment]
@dmcgowan : can you opine on this?

**Comment by dmcgowan** [comment]
I will only grudgingly support changes to this package which export previous non-exported functions. I don't see any issues sharing this function across packages though. If it gets too bad we can stick some of this stuff in another package too.

LGTM

**Comment by RichardScothern** [comment]
Can you rebase please @hopkings2008 ?

**Comment by hopkings2008** [comment]
@RichardScothern, @dmcgowan , thanks for taking time to checking this change. I have rebased my branch, but sadly, the jekins job failed at starting docker daemon:
+ docker --daemon --log-level=panic --storage-driver=aufs --exec-driver=native
main.mainDaemon()
	/go/src/github.com/docker/docker/docker/daemon.go:158 +0x134c
main.main()
	/go/src/github.com/docker/docker/docker/docker.go:93 +0x6b4

Seems that docker daemon has some problems in jekins env.

**Comment by hopkings2008** [comment]
@RichardScothern , i triggered the CI & jekins again, now all the tests are passed. Please help to check this change again. thanks in advance. :)

**Comment by hopkings2008** [comment]
@thaJeztah , i also rebase my changes to the latest one, and trigger the tests, then the janky failed at the same position:
"++ docker run -d -it --privileged -v /docker-versions.MXnkj/docker-master/docker:/usr/local/bin/docker -v /go/src/github.com/docker/distribution:/go/src/github.com/docker/distribution -e DOCKER_GRAPHDRIVER=aufs -e EXEC_DRIVER= distribution/docker-integration ./run_engine.sh"

According to @RichardScothern and @dmcgowan , my changes for the error message displaying is ok.

**Comment by stevvooe** [comment]
Closing in favor or https://github.com/docker/distribution/pull/1249.

**Pull request closed without merging**